newIDE/app/src/MainFrame/Toolbar/PreviewAndShareButtons.js

```diff
@@ -185,7 +187,11 @@ const PreviewAndShareButtons = React.memo<PreviewAndShareButtonsProps>(
       <LineStackLayout noMargin>
         <FlatButtonWithSplitMenu
           primary
-          onClick={onHotReloadPreview}
+          onClick={
+            !hasPreviewsRunning && preferences.values.takeScreenshotOnPreview // Take a screenshot on first preview.
```

Comment:
wondering if we should also take a screenshot only if:
- you don't have a thumbnailUrl already set (but the screenshots might be useful)
- you haven't taken a screenshot recently (to avoid taking one on every preview) - like every x minutes ?

Reply:
Yes every X minutes I think!

Reply:
added a 3 minute delay between each